gensim/test/test_fasttext.py

```diff
@@ -842,6 +842,176 @@ def test_sg_hs_against_wrapper(self):
+            "fox": [0.061692, 0.082914, 0.020081, -0.039159, 0.03296],
+            "jumps": [0.070107, 0.081465, 0.051763, 0.012084, 0.0050402],
+            "over": [0.055023, 0.03465, 0.01648, -0.11129, 0.094555],
+            "lazy": [-0.022103, -0.020126, -0.033612, -0.049473, 0.0054174],
+            "dog": [0.084983, 0.09216, 0.020204, -0.13616, 0.01118],
+        }
+        expected = {word: np.array(arr, dtype=np.float32) for word, arr in expected.items()}
+
+        for word, expected_vector in expected.items():
+            actual_vector = native.wv.word_vec(word)
+            self.assertTrue(np.allclose(expected_vector, actual_vector, atol=1e-1))
+
+    def test_sanity(self):
+        """Compare models trained on toy data.  They should be equal."""
+        trained = train_gensim()
+        native = load_native()
+
+        self.assertEqual(trained.bucket, native.bucket)
+
+        trained_vocab = {key: value.count for (key, value) in trained.wv.vocab.items()}
+        native_vocab = {key: value.count for (key, value) in native.wv.vocab.items()}
+        print('comparing vocab\ntrained:')
+        print(trained_vocab)
+        print('native:')
+        print(native_vocab)
+        self.assertEqual(trained_vocab, native_vocab)
+
+        #
+        # We do not compare most matrices directly, because they will never
+        # be equal unless many conditions are strictly controlled.
+        #
+        print_array(trained.wv.vectors_vocab, "trained.wv.vectors_vocab")
+        print_array(native.wv.vectors_vocab, "native.wv.vectors_vocab")
+        self.assertEqual(trained.wv.vectors_vocab.shape, native.wv.vectors_vocab.shape)
+        # self.assertTrue(np.array_equal(trained.wv.vectors_vocab, native.wv.vectors_vocab))
+
+        #
+        # Ensure the neural networks are identical for both cases.
+        #
+        trained_nn, native_nn = trained.trainables, native.trainables
+
+        print_array(trained_nn.syn1neg, "trained.syn1neg")
+        print_array(native_nn.syn1neg, "native.syn1neg")
+        self.assertEqual(trained_nn.syn1neg.shape, native_nn.syn1neg.shape)
+        #self.assertTrue(np.array_equal(trained_nn.syn1neg, native_nn.syn1neg))
+
+        print_array(trained_nn.vectors_lockf, "trained_nn.vectors_lockf")
+        print_array(native_nn.vectors_lockf, "native_nn.vectors_lockf")
+        self.assertEqual(trained_nn.vectors_lockf.shape, native_nn.vectors_lockf.shape)
+        self.assertTrue(np.array_equal(trained_nn.vectors_lockf, native_nn.vectors_lockf))
+
+        print_array(trained_nn.vectors_vocab_lockf, "trained_nn.vectors_vocab_lockf")
+        print_array(native_nn.vectors_vocab_lockf, "native_nn.vectors_vocab_lockf")
+        self.assertEqual(trained_nn.vectors_vocab_lockf.shape, native_nn.vectors_vocab_lockf.shape)
+        self.assertTrue(np.array_equal(trained_nn.vectors_vocab_lockf, native_nn.vectors_vocab_lockf))
+
+        #
+        # FIXME: Not sure why the values don't match
```

Comment:
to be sure, please check how they used (the answer should be in usage)

Reply:
Added comment to issue description